### PR TITLE
Add Support for Tekton Pipeline v1 APIs  

### PR DESCRIPTION
**Pull request by khrm**
# Changes
Add support for storing PipelineRun and TaskRun objects as v1.
These CRs are reconciled as the v1 version of Pipelines API and then
stored in DB as v1. Existing v1beta1 resources are kept as it is.
An ability to convert these older records to v1 will be added.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Records of TaskRuns and PipelineRuns are as stored as v1 in db instead of v1beta1.
Add support for storing PipelineRun and TaskRun objects as v1.
These CRs are reconciled as the v1 version of Pipelines API and then
stored in DB as v1. Existing v1beta1 resources are kept as it is.
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind feature

**Comment by khrm** [comment]
e2e tests are passing locally.  I upgraded kind version to resolve this.

**Review by khrm — COMMENTED** [review]
/hold cancel

**Comment by khrm** [comment]
@avinal @enarha @vdemeester 

Let's review and merge this.

**Comment by khrm** [comment]
@vdemeester @avinal @enarha Let's try to merge this.

**Review by vdemeester — APPROVED** [review]

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -45,9 +45,9 @@ import (
 	"os"
 	"path"
 
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	tektonv1beta1client "github.com/tektoncd/pipeline/pkg/client/clientset/versioned/typed/pipeline/v1beta1"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
```
Comment (enarha) [inline test/e2e/e2e_test.go:49]: the package is imported twice

**Review by enarha — COMMENTED** [review]
Apart from a nit, the rest is lgtm.

[EDIT] The documentation (./docs/*) still mentions "tekton.dev/v1beta1" (one example docs/api/README.md).

Furthermore, IMO the documentation could be improved by adding information about handling of existing v1beta1 data.

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (khrm) [reply test/e2e/e2e_test.go]: Fixed.

**Pull request merged**